src/Common/src/System/Net/Logging/HttpEventSource.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Tracing;
+using System.Net.Http;
+using System.Globalization;
+
+namespace System.Net
+{
+    [EventSource(Name = "Microsoft-System-Net-Http",
+        Guid = "bdd9a83e-1929-5482-0d73-2fe5e1c0e16d",
+        LocalizationResources = "FxResources.System.Net.Http.SR")]
+    internal sealed class HttpEventSource : EventSource
+    {
+        private const int ASSOCIATE_ID = 1;
+        private const int URI_BASE_ADDRESS_ID = 2;
+        private const int CONTENT_NULL_ID = 3;
+        private const int CLIENT_SEND_COMPLETED = 4;
+        private const int HEADERS_INVALID_VALUE_ID = 5;
+
+        private readonly static HttpEventSource s_log = new HttpEventSource();
+        private HttpEventSource() { }
+        public static HttpEventSource Log
+        {
+            get
+            {
+                return s_log;
+            }
+        }
+
+        internal static void Associate(object objA, object objB)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            s_log.Associate(LoggingHash.GetObjectName(objA),
+                            LoggingHash.HashInt(objA),
+                            LoggingHash.GetObjectName(objB),
+                            LoggingHash.HashInt(objB));
+        }
+
+        [Event(ASSOCIATE_ID, Keywords = Keywords.Default,
+            Level = EventLevel.Informational, Message = "[{0}#{1}]<-->[{2}#{3}]")]
+        internal unsafe void Associate(string objectA, int objectAHash, string objectB, int objectBHash)
+        {
+            const int SIZEDATA = 4;
```

Comment:
Nit: Naming convention: `sizeData`. (I think you can run the formatting tool to automatically correct these.)